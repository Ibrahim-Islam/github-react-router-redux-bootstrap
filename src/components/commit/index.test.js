import React from 'react'
import ReactDOM from 'react-dom'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom'
import { Commit } from './index'

configure({ adapter : new Adapter() })

const match = { params: { username: 'test', repo: 'test-repo' } }

describe('<Commit />', () => {

    it('should render without crashing', () => {
        const div = document.createElement('div')
        const mockGetCommits = jest.fn()
        const mockResetCommits = jest.fn()
        const mockSetError = jest.fn()
        ReactDOM.render(
        <BrowserRouter>
            <Commit error={''} 
                commitContainer={ {repo: '', commits: []} }
                match={ match }
                getCommits={mockGetCommits}
                resetCommits={mockResetCommits}
                setError={mockSetError} />
        </BrowserRouter>, div)
        expect(mockResetCommits).toHaveBeenCalledTimes(1)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('should match snapshot', () => {
        const mockGetCommits = jest.fn()
        const mockResetCommits = jest.fn()
        const mockSetError = jest.fn()
        const wrapper = shallow(
        <Commit error={''} 
            commitContainer={ {repo: '', commits: [{
                id: 1,
                headline: 'headline',
                message: 'message',
                date: 'date'
            }]} }
            match={ match }
            getCommits={mockGetCommits}
            resetCommits={mockResetCommits}
            setError={mockSetError} />
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should show alert if has error', () => {
        const mockGetCommits = jest.fn()
        const mockResetCommits = jest.fn()
        const mockSetError = jest.fn()
        const wrapper = mount(
            <BrowserRouter>
                <Commit error={ 'error '} 
                    commitContainer={ {repo: '', commits: [{
                        id: 1,
                        headline: 'headline',
                        message: 'message',
                        date: 'date'
                    }]} }
                    match={ match }
                    getCommits={mockGetCommits}
                    resetCommits={mockResetCommits}
                    setError={mockSetError} />         
            </BrowserRouter>
        )
        expect(wrapper.find('Alert').length).toBe(1)
        expect(wrapper.find('Table').length).toBe(0)
        expect(wrapper.find('div#preloader').length).toBe(0)
        wrapper.unmount()
    })

    it('should show loading', () => {
        const mockGetCommits = jest.fn()
        const mockResetCommits = jest.fn()
        const mockSetError = jest.fn()
        const wrapper = mount(
        <BrowserRouter>
            <Commit error={ 'error '} 
                commitContainer={ {repo: '', commits: [{
                    id: 1,
                    headline: 'headline',
                    message: 'message',
                    date: 'date'
                }]} }
                match={ match }
                getCommits={mockGetCommits}
                resetCommits={mockResetCommits}
                setError={mockSetError}
                showLoading={true} />    
        </BrowserRouter>
        )
        expect(wrapper.find('div#preloader').length).toBe(1)
        wrapper.unmount()
    })

    it('should get more commits if scrolled to bottom', () => {
        const mockGetCommits = jest.fn()
        const mockResetCommits = jest.fn()
        const mockSetError = jest.fn()
        
        const map = {};
        window.addEventListener = jest.fn((event, cb) => {
            map[event] = cb;
        });
        
        const wrapper = shallow(<Commit 
                error={''} 
                commitContainer={ 
                    {
                        repo: '', 
                        pageInfo: { 
                            hasNextPage: true
                        },
                        commits: [{
                            id: 1,
                            headline: 'headline',
                            message: 'message',
                            date: 'date'
                        }] 
                    } 
                }
                match={ match }
                getCommits={mockGetCommits}
                resetCommits={mockResetCommits}
                setError={mockSetError}
                showLoading={false} />)

        global.innerHeight = 0
        map.scroll()
        
        expect(mockResetCommits).toHaveBeenCalledTimes(1)
        expect(mockGetCommits).toHaveBeenCalledTimes(1)
    })

})