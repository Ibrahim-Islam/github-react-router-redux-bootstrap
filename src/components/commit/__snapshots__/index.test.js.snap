// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Commit /> should match snapshot 1`] = `
<div
  className="gh-table-body"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <Breadcrumb
          aria-label="breadcrumb"
          listTag="ol"
          tag="nav"
        >
          <BreadcrumbItem
            tag="li"
          >
            <Link
              replace={false}
              to="/"
            >
              Search
            </Link>
          </BreadcrumbItem>
          <BreadcrumbItem
            tag="li"
          >
            test
          </BreadcrumbItem>
          <BreadcrumbItem
            tag="li"
          >
            <Link
              replace={false}
              to="/test/repositories"
            >
              Repositories
            </Link>
          </BreadcrumbItem>
          <BreadcrumbItem
            tag="li"
          >
            test-repo
          </BreadcrumbItem>
          <BreadcrumbItem
            active={true}
            tag="li"
          >
            Commits
          </BreadcrumbItem>
        </Breadcrumb>
      </div>
    </div>
    <div>
      <div
        className="row"
      >
        <div
          className="col-lg-8"
        >
          <h3>
            test-repo
          </h3>
        </div>
        <div
          className="col-lg-4"
        >
          <Input
            className="gh-search"
            onChange={[Function]}
            placeholder="Enter text to search...."
            type="text"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <ListGroup
            tag="ul"
          >
            <CommitItem
              date="date"
              headline="headline"
              id={1}
              key="1"
              message="message"
            />
          </ListGroup>
        </div>
      </div>
    </div>
  </div>
</div>
`;
